framework/data/Sort.php

```diff
@@ -229,7 +229,7 @@ public function getOrders($recalculate = false)
                     $orders[$name] = $dir;
                 }
             } else {
-                $orders[] = $columns;
+                $orders = $columns;
```

Comment:
in this case `columns` is not an array. The return type of this method is `array`, so assigning columns to orders does not make sense.